etcd/watchmanager/types.go

```diff
@@ -35,7 +35,7 @@ type WatchManager interface {
 }
 
 // UpdateFn is called when an event on the watch channel happens
-type UpdateFn func(key string) error
+type UpdateFn func(key string, events []*clientv3.Event) error
```

Comment:
seems you are treating nil slice and empty slice the same right now? you can still keep this logic in updateFn? Also Do you know what's contained in the watch response when a progress watch update happens?